trie/zk_trie_test.go

```diff
@@ -264,3 +268,72 @@ func TestZkTrieDelete(t *testing.T) {
 		assert.Equal(t, hashes[i].Hex(), hash.Hex())
 	}
 }
+
+func TestEquivalence(t *testing.T) {
+	t.Skip()
+
+	zkDb, err := leveldb.New("/Users/omer/Documents/go-ethereum/l2geth-datadir/geth/chaindata", 0, 0, "", true)
+	require.NoError(t, err)
+	mptDb, err := leveldb.New("/Users/omer/Documents/go-ethereum/l2geth-datadir-mpt/geth/chaindata", 0, 0, "", true)
+	require.NoError(t, err)
+
+	zkRoot := common.HexToHash("0x294b458b5b571bb634dbe9a81331dd2aabb5ef40cdc0328b075a9666d5df55d0")
+	mptRoot, err := rawdb.ReadDiskStateRoot(mptDb, zkRoot)
+	require.NoError(t, err)
+
+	checkTrieEquality(t, &dbs{
+		zkDb:  zkDb,
+		mptDb: mptDb,
+	}, zkRoot, mptRoot, checkAccountEquality)
+}
+
+type dbs struct {
+	zkDb  *leveldb.Database
+	mptDb *leveldb.Database
+}
+
+var accountsLeft = -1
+
+func checkTrieEquality(t *testing.T, dbs *dbs, zkRoot, mptRoot common.Hash, leafChecker func(*testing.T, *dbs, []byte, []byte)) {
+	zkTrie, err := NewZkTrie(zkRoot, NewZktrieDatabase(dbs.zkDb))
+	require.NoError(t, err)
+
+	mptTrie, err := NewSecure(mptRoot, NewDatabaseWithConfig(dbs.mptDb, &Config{Preimages: true}))
+	require.NoError(t, err)
+
+	expectedLeaves := zkTrie.CountLeaves()
```

Comment:
should the leaf count for the zkTrie be equal to the leaf count of the mptTrie?

Reply:
Yeap, pretty much. Accounts and storage variables are stored in the leaves of both type of nodes.